### PR TITLE
Use organization workflow to build and push images

### PR DESCRIPTION
**Pull request by domire8**
## Description

- https://github.com/aica-technology/.github/issues/3

In this PR, I'm using the organization workflows to create the component-sdk images. This reduces code duplication and IMO streamlines the local workflows here more. As a reminder, we have 4 workflows here:
1. manual-dispatch: build an image of your choice (base tag and output tag)
2. check-upstream: scheduled check if the ros2-modulo images have changed
3. build-push-develop: on push on develop, rebuild the latest and devel image
4. build-push-tagged: on publish of a new tag, rebuild a tagged image.

Note that for the `latest` and `devel` image, the component-sdk branch is `develop `, whereas for the tagged image, we take `main` branch. I think this still makes sense.

## Supporting information
This has been tested on my fork

## Review guidelines

Estimated Time of Review: 15 minutes
Not sure if the diff here makes a lot of sense. It might be nice to just look at the files like that.

### CODE REVIEW

**Comment by domire8** [comment]
This is also back under revision

**Comment by domire8** [comment]
And we're back to ready for review. Was actually quite nice to tag the workflow and allow default values for the two inputs :+1:

File: .github/workflows/check-upstream.yml [hunk 1]
```diff
@@ -13,94 +13,89 @@ jobs:
 
   check-hash:
```
Comment (eeberhard) [inline .github/workflows/check-upstream.yml:14]: Just a thought, maybe a "check hash" workflow can be the next todo item for the .github repository.

**Review by eeberhard — APPROVED** [review]
I'm not sure how thorough I can be but it seems clean to me. A definite improvement to use the tagged centralized workflows.

**Comment by domire8** [comment]
> I'm not sure how thorough I can be but it seems clean to me.

Yes I understand and it is why I'm reluctant to open the biggest one of those PRs, the one in docker images. It's going to be a big diff but the functionality stays the same so its hard to review. But I try to test those changes as good as possible in my forks, so apart from potential minor bugs, it should be good..

**Pull request merged**